definitions/checks/check_available_memory.rb

```diff
@@ -0,0 +1,29 @@
+class Checks::CheckAvailableMemory < ForemanMaintain::Check
+  metadata do
+    label :check_available_memory
+    description 'Check if system has sufficient RAM available'
+    tags :pre_upgrade
+    confine do
+      feature(:downstream) || feature(:upstream)
+    end
+  end
+
+  def run
+    system_memory_check
+  end
+
+  def system_memory_check
+    katello_min_ram = 7_900_000
+    foreman_min_ram = 4_100_000
+    total_ram = `grep MemTotal /proc/meminfo | awk '{print $2}'`.to_i
+    if (feature(:upstream) && katello_plugin_enabled?) || feature(:downstream)
+      assert(katello_min_ram < total_ram, 'This system has less than 8 GB of RAM.')
+    elsif feature(:upstream) && !katello_plugin_enabled?
+      assert(foreman_min_ram < total_ram, 'This system has less than 4 GB of RAM.')
+    end
+  end
+
+  def katello_plugin_enabled?
+    Features::Installer.new.answers.key? 'katello'
```

Comment:
Instead of `Features::Installer.new`, you can directly use `feature(:installer)`